lib/active_fedora/associations/builder/association.rb

```diff
@@ -10,7 +10,7 @@ class << self
     # configure_dependency
     def self.build(model, name, options, &block)
       if model.dangerous_attribute_method?(name)
-        Deprecation.warn(ActiveFedora::Base, "You tried to define an association named #{name} on the model #{model.name}, but " \
+        ActiveFedora::Base.logger.error("You tried to define an association named #{name} on the model #{model.name}, but " \
```

Comment:
I'm not sure that a deprecation is better. Deprecation seems incorrect as this is not a deprecation. Are you suggesting that we raise an error?